plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/NameEmbeddings.kt

```diff
@@ -31,19 +33,51 @@ data class ClassName(val packageName: FqName, val className: Name) : MangledName
         get() = "pkg_${packageName.asString()}\$class_${className.asString()}"
 }
 
-data class ClassMemberName(val className: ClassName, val name: Name) : MangledName {
+open class ClassMemberName(val className: ClassName, val name: Name) : MangledName {
```

Comment:
I think we always want `MangledName`s to be data classes since we want equality be derived from the name fields instead of referential equality. To make sure no one instantiates this directly make it an interface or an abstract class.